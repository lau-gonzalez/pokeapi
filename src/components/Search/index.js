import React from "react";

const Search = ({ setValue, placeholder }) => {
  const handleChange = (e) => setValue(e.target.value.toLowerCase());
  return (
    <div className="input-group" style={{ width: 250 }}>
      <input
        type="text"
        className="form-control"
        placeholder={placeholder}
        onChange={handleChange}
      />
    </div>
  );
};

export default Search;
