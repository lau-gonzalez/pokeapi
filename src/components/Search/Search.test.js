import { render, cleanup, screen } from "@testing-library/react";
import Search from "./index";

afterEach(cleanup);

const setValue = jest.fn();

describe("<Search/>", () => {
  it("renders the component", () => {
    const { asFragment } = render(
      <Search setValue={setValue} placeholder="Search something" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the input", () => {
    render(<Search setValue={setValue} placeholder="Search something" />);
    expect(screen.getByPlaceholderText("Search something")).toBeInTheDocument();
  });
});
