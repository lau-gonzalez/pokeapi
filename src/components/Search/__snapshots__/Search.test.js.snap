// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> renders the component 1`] = `
<DocumentFragment>
  <div
    class="input-group"
    style="width: 250px;"
  >
    <input
      class="form-control"
      placeholder="Search something"
      type="text"
    />
  </div>
</DocumentFragment>
`;
